testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/LoginPasswordResetPage.java

```diff
@@ -62,8 +66,20 @@ public String getSuccessMessage() {
         return emailSuccessMessage != null ? emailSuccessMessage.getText() : null;
     }
 
+    public String getUsernameError() {
+        try {
+            return usernameError != null ? usernameError.getText() : null;
+        } catch (NoSuchElementException e) {
+            return null;
+        }
+    }
+
     public String getErrorMessage() {
-        return emailErrorMessage != null ? emailErrorMessage.getText() : null;
+        try {
+            return emailErrorMessage != null ? emailErrorMessage.getText() : getUsernameError();
```

Comment:
The test should be deterministic and should know whether it's fetching the error message from the notification or from the field.